cmd/prchecker/main_test.go

```diff
@@ -0,0 +1,232 @@
+	testname := fmt.Sprintf("validate get commit changes converts to changeset")
+
+	t.Run(testname, func(t *testing.T) {
+
+		client := FakeGitHubService{
+			commit: &github.RepositoryCommit{
+				SHA:   github.String("abc123"),
+				Files: []github.CommitFile{{Filename: github.String("foo")}, {Filename: github.String("bar")}},
+			},
+		}
+
+		result, _ := GetCommitChanges(&client, "abc123")
+
+		if result.id != "abc123" {
+			t.Errorf("got %v, want %v", result.id, "abc123")
+		}
+
+		if len(result.files) != 2 {
+			t.Errorf("got %v, want %v", len(result.files), "2")
+		}
+
+		expectedFiles := []string{"foo", "bar"}
+		if !StringSliceAreEqual(result.files, expectedFiles) {
+			t.Errorf("got %v, want %v", result.files, expectedFiles)
+		}
+	})
+}
+
+func TestGetPullRequestCommits(t *testing.T) {
+	testname := fmt.Sprintf("validate get commit changes converts to changeset")
+
+	t.Run(testname, func(t *testing.T) {
+
+		client := FakeGitHubService{
+			commit: &github.RepositoryCommit{
+				SHA:   github.String("abc123"),
+				Files: []github.CommitFile{{Filename: github.String("foo")}, {Filename: github.String("bar")}},
+			},
+			commitList: []*github.RepositoryCommit{
+				&github.RepositoryCommit{
+					SHA: github.String("abc123"),
+				},
+				&github.RepositoryCommit{
+					SHA: github.String("abc123"),
+				},
+			},
+		}
+
+		result, _ := GetPullRequestCommits(&client, 42)
+
+		if len(result) != 2 {
+			t.Errorf("got %v, want %v", len(result), 2)
+		}
+
+		expectedFiles := []string{"foo", "bar"}
+		if !StringSliceAreEqual(result[0].files, expectedFiles) {
+			t.Errorf("[%d] got %v, want %v", 0, result[0].files, expectedFiles)
+		}
+		if !StringSliceAreEqual(result[1].files, expectedFiles) {
+			t.Errorf("[%d] got %v, want %v", 1, result[1].files, expectedFiles)
+		}
+	})
+}
+
+func TestContstructingChangeset(t *testing.T) {
+	var tests = []struct {
+		sha      string
+		files    []github.CommitFile
+		expected Changeset
+	}{
+		{
+			"abc123",
+			[]github.CommitFile{{Filename: github.String("foo")}},
+			Changeset{id: "abc123", files: []string{"foo"}},
+		},
+		{
+			"1234",
+			[]github.CommitFile{{Filename: github.String("foo")}, {Filename: github.String("bar")}},
+			Changeset{id: "1234", files: []string{"foo", "bar"}},
+		},
+	}
+
+	for _, tt := range tests {
+		testname := fmt.Sprintf("is %v in %v", tt.sha, tt.expected.id)
+
+		t.Run(testname, func(t *testing.T) {
+			commit := &github.RepositoryCommit{SHA: github.String(tt.sha), Files: tt.files}
+			result := ConstructChangeset(commit)
+
+			if tt.expected.id != result.id {
+				t.Errorf("got %v, want %v", result.id, tt.expected.id)
+			}
+
+			if !StringSliceAreEqual(tt.expected.files, result.files) {
+				t.Errorf("got %v, want %v", result.files, tt.expected.files)
+			}
+		})
+	}
+}
+
+func TestIsChangesetSpanning(t *testing.T) {
+	var tests = []struct {
+		changeset []string
+		files     []string
+		expected  bool
+	}{
+		{[]string{"1"}, []string{"1"}, false},
```

Comment:
Nit: I usually find it easier to read if we don't put multiple fields on one line. I like the pattern from the previous test better.

Reply:
Would you also apply this [here](https://github.com/kubernetes-sigs/kustomize/pull/3139/files/a9f5d34579ae633208aebd034504e9a666c0a293#diff-9989dd5b0b267de296fb7f62a9b853c97e15dc75a6079f2ace3f23059bdfc043R46) where only two parameters are provided?